viabel/optimization_diagnostics.py

```diff
@@ -156,16 +221,25 @@ def autocorrelation(iterate_chains, warmup=500, param_idx=0, lag_max=80):
     return neff, rho_t, autocov, np.asarray(a)
 
 
-def compute_khat_iterates(iterate_chains, warmup=500, param_idx=0, increasing= True, fraction=0.15):
-    '''
-    function computes the khat for iterates of VI, preferable to run it after approximate convergence .
-    :param iterate_chains:
-    :param warmup:
-    :param param_idx:
-    :param increasing:
-    :param fraction:
-    :return:
-    '''
+def compute_khat_iterates(iterate_chains, warmup=500, param_idx=0, increasing=True, fraction=0.15):
+    """
+    Compute the khat over iterates for a variational parameter after removing warmup.
+    Parameters
+    ----------
+    iterate_chains : multi-dimensional array, shape=(n_chains, n_iters, n_var_params)
+
+    warmup : warmup iterates
+
+    param_idx : index of the variational parameter
+
+    increasing : boolean sort array in increasing order: TRUE or decreasing order:FALSE
+
+    fraction: the fraction of iterates
```

Comment:
Why have warmup and fraction? It doesn't seem to fundamentally provide any extra flexibility

Reply:
this is fixed now.